matlab/src/cpp/arrow/matlab/tabular/proxy/schema.cc

```diff
@@ -0,0 +1,196 @@
+        using FieldProxy = arrow::matlab::type::proxy::Field;
+        mda::ArrayFactory factory;
+
+        mda::StructArray args = context.inputs[0];
+        const mda::TypedArray<int32_t> index_mda = args[0]["Index"];
+        const auto matlab_index = int32_t(index_mda[0]);
+        // Note: MATLAB uses 1-based indexing, so subtract 1.
+        // arrow::Schema::field does not do any bounds checking.
+        const int32_t index = matlab_index - 1;
+        const auto num_fields = schema->num_fields();
+
+        if (num_fields == 0) {
+            const auto& error = makeEmptySchemaError();
+            context.error = error;
+            return;
+        }
+
+        if (matlab_index < 1 || matlab_index > num_fields) {
+            using namespace libmexclass::error;
+            const std::string& error_message_id = std::string{error::ARROW_TABULAR_SCHEMA_INVALID_NUMERIC_FIELD_INDEX};
+            std::stringstream error_message_stream;
+            error_message_stream << "Invalid field index: ";
+            error_message_stream << matlab_index;
+            error_message_stream << ". Field index must be between 1 and the number of fields (";
+            error_message_stream << num_fields;
+            error_message_stream << ").";
+            const std::string& error_message = error_message_stream.str();
+            context.error = Error{error_message_id, error_message}; 
+            return;
+        }
+
+        const auto& field = schema->field(index);
+        auto field_proxy = std::make_shared<FieldProxy>(field);
+        const auto field_proxy_id = ProxyManager::manageProxy(field_proxy);
+        const auto field_proxy_id_mda = factory.createScalar(field_proxy_id);
+
+        context.outputs[0] = field_proxy_id_mda;
+    }
+
+    void Schema::getFieldByName(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        using namespace libmexclass::proxy;
+        using FieldProxy = arrow::matlab::type::proxy::Field;
+        mda::ArrayFactory factory;
+
+        mda::StructArray args = context.inputs[0];
+        const mda::StringArray name_mda = args[0]["Name"];
+        const auto name_utf16 = std::u16string(name_mda[0]);
+        MATLAB_ASSIGN_OR_ERROR_WITH_CONTEXT(const auto name, arrow::util::UTF16StringToUTF8(name_utf16), context, error::UNICODE_CONVERSION_ERROR_ID);
+        const std::vector<std::string> names = {name};
+        MATLAB_ERROR_IF_NOT_OK_WITH_CONTEXT(schema->CanReferenceFieldsByNames(names), context, error::ARROW_TABULAR_SCHEMA_AMBIGUOUS_FIELD_NAME);
+        const auto field = schema->GetFieldByName(name);
+        if (!field) {
+            // Note: This line should never be reached because CanReferenceFieldsByNames
+            //       should already handle validating whether the supplied field name is valid.
+            const auto& error = makeUnknownFieldNameError(name);
+            context.error = error;
+            return;
+        }
```

Comment:
Can we remove this?

Reply:
I was slightly worried about the possibility of a `nullptr` being returned here for some unexpected reason, but this check isn't strictly necessary since we already validate that the field name is valid via `CanReferenceFieldsByNames`.

I'll remove this check.

Reply:
Fixed.